overlord/snapstate/snapmgr.go

```diff
@@ -1302,9 +1303,14 @@ func (m *SnapManager) ensureMountsUpdated() error {
 			//   This is especially relevant for the snapd snap as if
 			// this happens, it would end up in a bad state after
 			// an update.
+			// TODO refactor so the check for kernel type is not repeated
+			// in the installation case
+			snapType, _ := snapSt.Type()
 			if _, err = sysd.EnsureMountUnitFile(info.MountDescription(),
 				squashfsPath, whereDir, "squashfs",
-				systemd.EnsureMountUnitFlags{PreventRestartIfModified: true}); err != nil {
+				systemd.EnsureMountUnitFlags{
+					PreventRestartIfModified: true,
+					StartBeforeDriversLoad:   snapType == snap.TypeKernel}); err != nil {
```

Comment:
not quite related to the PR but do we need to add code here about components mounts as well?

Reply:
Right, I've added a TODO about that